pylint/checkers/classes.py

```diff
@@ -649,29 +682,41 @@ def visit_functiondef(self, node):
         if not node.is_method():
             return
 
-        self._check_useless_super_delegation(node)
-
         klass = node.parent.frame()
         self._meth_could_be_func = True
         # check first argument is self if this is actually a method
         self._check_first_arg_for_type(node, klass.type == 'metaclass')
         if node.name == '__init__':
+            self._check_useless_super_delegation(node)
             self._check_init(node)
             return
-        # check signature if the method overloads inherited method
-        for overridden in klass.local_attr_ancestors(node.name):
-            # get astroid for the searched method
-            try:
-                meth_node = overridden[node.name]
-            except KeyError:
-                # we have found the method but it's not in the local
-                # dictionary.
-                # This may happen with astroid build from living objects
-                continue
-            if not isinstance(meth_node, astroid.FunctionDef):
-                continue
-            self._check_signature(node, meth_node, 'overridden', klass)
-            break
+
+        # if the method overloads inherited method then
+        # - check for useless super delegation if the argument default values are
+        #   the same in the base method and in the overridden one
+        # - check signature
+        # otherwise just check for useless super delegation
+        overridden_methods = list(klass.local_attr_ancestors(node.name))
+        if overridden_methods:
+            for overridden in overridden_methods:
+                # get astroid for the searched method
+                try:
+                    meth_node = overridden[node.name]
+                except KeyError:
+                    # we have found the method but it's not in the local
+                    # dictionary.
+                    # This may happen with astroid build from living objects
+                    continue
+                if not isinstance(meth_node, astroid.FunctionDef):
+                    continue
+                # Don't care about functions with unknown argument (builtins).
+                if not _has_different_parameters_default_value(meth_node.args, node.args):
```

Comment:
I'm not exactly sure this makes sense. Basically we can potentially call ``_check_useless_super_delegation`` for each overridden method, which can be many in certain rich hierarchy classes. This makes me think that the ``_has_different_parameters_default_value`` shouldn't actually be here, but rather in ``_check_useless_super_delegation`` itself

Reply:
In fact, i put it here to avoid code duplicate but you are definitely right it make more sense to put it inside ``_check_useless_super_delegation`` itself.